### PR TITLE
Remove legacy widget area constants.

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #5476 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/ModulesList.js [hunk 1]
```diff
@@ -1,116 +0,0 @@
-/**
- * ModulesList component.
```
Comment (hussain-t) [inline assets/js/components/ModulesList.js:2]: Removing this file seems to be an unrelated change to this issue.

File: assets/js/components/ModulesListItem.js [hunk 2]
```diff
@@ -1,93 +0,0 @@
-/**
- * ModulesListItem component.
```
Comment (hussain-t) [inline assets/js/components/ModulesListItem.js:2]: Same here, removing this file seems to be an unrelated change to this issue.

File: assets/js/googlesitekit/widgets/register-defaults.js [hunk 3]
```diff
@@ -42,16 +42,12 @@ const { AREA_DASHBOARD_ACQUISITION, ...ADDITIONAL_WIDGET_AREAS } = WIDGET_AREAS;
...
-			width: [ WIDGET_WIDTHS.HALF, WIDGET_WIDTHS.FULL ],
-			Component: URLSearchWidget,
-			wrapWidget: false,
-		},
-		[ AREA_DASHBOARD_ACQUISITION ]
-	);
+	widgetsAPI.registerWidget( WIDGET_URL_SEARCH, {
+		priority: 100,
+		width: [ WIDGET_WIDTHS.HALF, WIDGET_WIDTHS.FULL ],
+		Component: URLSearchWidget,
```
Comment (hussain-t) [inline assets/js/googlesitekit/widgets/register-defaults.js:48]: The `URLSearchWidget` component has to be deleted. Please refer back to the IB.

File: stories/notifications.stories.js [hunk 4]
```diff
@@ -28,71 +28,14 @@ import { __ } from '@wordpress/i18n';
...
 import { CORE_USER } from '../assets/js/googlesitekit/datastore/user/constants';
-import { MODULES_ADSENSE } from '../assets/js/modules/adsense/datastore/constants';
-import { CORE_MODULES } from '../assets/js/googlesitekit/modules/datastore/constants';
-import { withConnected } from '../assets/js/googlesitekit/modules/datastore/__fixtures__';
-import SuccessGreenSVG from '../assets/svg/graphics/success-green.svg';
 import AwardSVG from '../assets/svg/graphics/award.svg';
 
 storiesOf( 'Global/Notifications', module )
-	.add(
-		'Module Setup Complete',
```
Comment (hussain-t) [inline stories/notifications.stories.js:49]: Why do we need to remove this story?

File: stories/modules-list.stories.js [hunk 5]
```diff
@@ -1,45 +0,0 @@
-/**
- * Modules List stories.
```
Comment (hussain-t) [inline stories/modules-list.stories.js:2]: Same here; this seems to be unrelated.

**Review by hussain-t — CHANGES_REQUESTED** [review]
@sashadoes, there are a few points to be considered:
- I can see unrelated changes to this ticket in this PR. All the changes related to `ModuleList*` are unnecessary. 
- Some of the changes mentioned in the IB weren't found. For example, deleting the related files.
- Could you add more details in the QAB, even if it's the same as the AC, that would help the QA to carry out the testing?

Thanks!

File: .storybook/storybook-data.js [hunk 6]
```diff
@@ -80,19 +80,6 @@ module.exports = [
...
-		name: 'Module Setup Complete',
-		story: 'Module Setup Complete',
-		parameters: {
-			fileName: './stories/notifications.stories.js',
-			options: {
-				hierarchyRootSeparator: '|',
-				hierarchySeparator: {},
-			},
-		},
-	},
```
Comment (hussain-t) [inline .storybook/storybook-data.js:95]: Why do we have to delete these lines?

**Review by hussain-t — CHANGES_REQUESTED** [review]
All looks good except the deletion in the `.storybook/storybook-data.js` file. @sashadoes please check my comment. Thanks!

**Review by hussain-t — APPROVED** [review]
LGTM 👍

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Pull request merged**